### PR TITLE
[WIP]Pytest Style analysis/test_contacts.py

### PR DESCRIPTION
**Pull request by utkbansal**
Fixes #

Changes made in this Pull Request:
 - 

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

File: testsuite/MDAnalysisTests/analysis/test_contacts.py [hunk 1]
```diff
@@ -87,18 +79,18 @@ def test_soft_cut_q_unfolded():
...
-    assert_equal(contacts.hard_cut_q([1], 2), 1)
-    assert_equal(contacts.hard_cut_q([2], 1), 0)
-    assert_equal(contacts.hard_cut_q([2, 0.5], 1), 0.5)
-    assert_equal(contacts.hard_cut_q([2, 3], [3, 4]), 1)
-    assert_equal(contacts.hard_cut_q([4, 5], [3, 4]), 0)
+    assert contacts.hard_cut_q([1], 2) == 1
+    assert contacts.hard_cut_q([2], 1) == 0
+    assert contacts.hard_cut_q([2, 0.5], 1) == 0.5
+    assert contacts.hard_cut_q([2, 3], [3, 4]) == 1
+    assert contacts.hard_cut_q([4, 5], [3, 4]) == 0
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/analysis/test_contacts.py]: Parametrize

File: testsuite/MDAnalysisTests/analysis/test_contacts.py [hunk 2]
```diff
@@ -87,18 +79,18 @@ def test_soft_cut_q_unfolded():
...
 
 
 def test_radius_cut_q():
     # check some extremal points
-    assert_equal(contacts.radius_cut_q([1], None, 2), 1)
-    assert_equal(contacts.radius_cut_q([2], None, 1), 0)
-    assert_equal(contacts.radius_cut_q([2, 0.5], None, 1), 0.5)
+    assert contacts.radius_cut_q([1], None, 2) == 1
+    assert contacts.radius_cut_q([2], None, 1) == 0
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/analysis/test_contacts.py]: Parametrize

File: testsuite/MDAnalysisTests/analysis/test_contacts.py [hunk 3]
```diff
@@ -21,30 +21,22 @@
...
 
 from MDAnalysisTests import tempdir
 
 
 def test_soft_cut_q():
     # just check some of the extremal points
-    assert_equal(contacts.soft_cut_q([0], [0]), .5)
+    assert contacts.soft_cut_q([0], [0]) == .5
     assert_almost_equal(contacts.soft_cut_q([100], [0]), 0)
     assert_almost_equal(contacts.soft_cut_q([-100], [0]), 1)
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/analysis/test_contacts.py]: Parametrize

File: testsuite/MDAnalysisTests/analysis/test_contacts.py [hunk 4]
```diff
@@ -151,50 +143,50 @@ def soft_cut(ref, u, selA, selB, radius=4.5, beta=5.0, lambda_constant=1.8):
...
+        return mda.Universe(PSF, DCD)
 
-    def tearDown(self):
-        # reset trajectory
-        self.universe.trajectory[0]
-        del self.universe
+    @staticmethod
+    @pytest.fixture()
+    def trajectory(universe):
+        return universe.trajectory
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/analysis/test_contacts.py]: No fixture needed

File: testsuite/MDAnalysisTests/analysis/test_contacts.py [hunk 5]
```diff
@@ -151,50 +143,50 @@ def soft_cut(ref, u, selA, selB, radius=4.5, beta=5.0, lambda_constant=1.8):
...
 
-    def _run_Contacts(self, **kwargs):
-        acidic = self.universe.select_atoms(self.sel_acidic)
-        basic = self.universe.select_atoms(self.sel_basic)
+    def _run_Contacts(self, universe, **kwargs):
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/analysis/test_contacts.py]: This should rather be a fixture

File: testsuite/MDAnalysisTests/analysis/test_contacts.py (hunk 3 above)
Comment (utkbansal) [inline testsuite/MDAnalysisTests/analysis/test_contacts.py]: @kain88-de Should I break this into two functions? One with `assert` and another with `assert_almost_equal` which would be parametrized?

File: testsuite/MDAnalysisTests/analysis/test_contacts.py (hunk 5 above)
Comment (utkbansal) [inline testsuite/MDAnalysisTests/analysis/test_contacts.py]: Not sure how I'd pass it `**kwargs` then.

File: testsuite/MDAnalysisTests/analysis/test_contacts.py (hunk 3 above)
Comment (kain88-de) [inline testsuite/MDAnalysisTests/analysis/test_contacts.py]: they can all use assert_almost_equal

Comment (utkbansal) [inline testsuite/MDAnalysisTests/analysis/test_contacts.py]: Doesn't sound like a good idea.

**Pull request merged**